src/coins.cpp

```diff
@@ -321,8 +315,8 @@ void CCoinsViewCache::SanityCheck() const
         if (entry.IsDirty()) attr |= 1;
         if (entry.IsFresh()) attr |= 2;
         if (entry.coin.IsSpent()) attr |= 4;
-        // Only 5 combinations are possible.
-        assert(attr != 2 && attr != 4 && attr != 7);
+        // Only 4 combinations are possible.
+        assert(attr != 2 && attr != 4 && attr != 6 && attr != 7);
```

Comment:
great, now that we have this, can we call `SanityCheck` in more tests? I understood that wasn't the case since we still had invalid states in fuzz/unit tests

Reply:
We can't since we are still calling this with spent, dirty, and fresh entries. These would be deleted in production code. That can also be tackled in a follow-up.

Reply:
You did enable it for a few additional cases, that's already a win 👍

Reply:
Err, I did but didn't test locally before pushing. I will have to revert. Tests are failing.

Reply:
I can make a follow-up to remove spent fresh and dirty entries from the test cases as well. I don't think that should happen in this PR though to keep it focused.
The previous PR also did this.